### PR TITLE
Fix PluginCache parsing errors for standard and custom plugins

### PR DESCRIPTION
**Pull request by htsukamoto5**
## Summary

- Fixes two root causes behind the "* Error parsing [plugin]" messages appearing in the terminal
- The data block extraction regex was overshooting into the rest of the `info` object, causing throws on plugins with nested `data` like `jsPsych-instructions` — replaced with brace counting
- Non-ok HTTP responses from unpkg (e.g. for custom or unknown plugins) were being passed to the parser as if they were valid source — added an HTTP status check that falls back to `"unknown"` and warns cleanly instead of throwing
- Four new tests in `plugin-cache.test.ts` cover flat plugins, nested plugins, nested `view_history`, and 404 fallback behaviour

This branch is independent from #46, #48, and #49 and can be reviewed and merged on its own.

> **Note:** Once this merges, some variables that PR #48's interactive prompt would have flagged as `"unknown"` will now be correctly resolved from plugin source automatically, reducing the number of prompts users see.

## Test plan

- [ ] Run `npm test` in `packages/metadata` — all tests should pass
- [ ] Run CLI against a data directory that includes `jsPsych-instructions` trials — confirm no "Error parsing" messages appear and `view_history` gets a real description

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/metadata/tests/plugin-cache.test.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+const info = <const>{
+  name: "mock-nested",
+  parameters: {},
+  data: {
+    /** History of pages viewed. */
+    view_history: {
+      type: ParameterType.COMPLEX,
+      array: true,
+      nested: {
+        page_index: {
```
Comment (jodeleeuw) [inline packages/metadata/tests/plugin-cache.test.ts:34]: should we have jsdoc comments above these nested parameters, and see if those get extracted into the description when the column is expanded?

something like:
`/** The page index in the instructions */`

Comment (htsukamoto5) [reply packages/metadata/tests/plugin-cache.test.ts:34]: Update on PR #50: 

Addressed your suggestion by adding JSDoc (/** The page index in the instructions */ and /** Time spent viewing the page in milliseconds */) to the nested params (page_index, viewing_time), and wrote a test for it. But this actually revealed another parser bug where page_index (the first nested param) wasn't being extracted at all as its own cache entry. The non-greedy regex was stopping at the first }, it found after view_history: {, which happened to be inside page_index's block, so page_index got consumed as part of view_history's props and so was never extracted. viewing_time was accidentally working because it appeared after the first },.

Fixed it by replacing the flat regex in parseJavadocString with a new extractJsdocFields method that uses brace counting to find each variable's true closing brace, then recursively processes any nested: sub-object.



Also got a code review from another instance of Claude and addressed all the feedback: advanced lastIndex past each variable's closing brace to prevent double-processing, tightened a test assertion that could pass vacuously, added an edge-case test for plugins with no data: block, documented known parser limitations, extracted a shared findMatchingBrace helper to eliminate duplicated brace-counting logic, added a continue guard for unbalanced source, and tightened the property regex. 

All 25 tests pass

**Review by jodeleeuw — APPROVED** [review]
